pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/GetOptionsForActivityInstanceQuestionRoute.java

```diff
@@ -0,0 +1,121 @@
+
+/**
+ * This route returns the list of suggestions for the supplied Picklist Option label
+ */
+public class GetOptionsForActivityInstanceQuestionRoute implements Route {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GetOptionsForActivityInstanceQuestionRoute.class);
+    private static final int DEFAULT_LIMIT = 100;
+    private final I18nContentRenderer renderer;
+
+    public GetOptionsForActivityInstanceQuestionRoute(I18nContentRenderer renderer) {
+        this.renderer = renderer;
+    }
+
+    @Override
+    public RemoteAutoCompleteResponse handle(Request request, Response response) {
+        String studyGuid = request.params(RouteConstants.PathParam.STUDY_GUID);
+        String participantGuid = request.params(RouteConstants.PathParam.USER_GUID);
+        String instanceGuid = request.params(RouteConstants.PathParam.INSTANCE_GUID);
+        String questionStableId = request.params(RouteConstants.PathParam.QUESTION_STABLE_ID);
+
+        DDPAuth ddpAuth = RouteUtil.getDDPAuth(request);
+        String operatorGuid = StringUtils.defaultIfBlank(ddpAuth.getOperator(), participantGuid);
+        boolean isStudyAdmin = ddpAuth.hasAdminAccessToStudy(studyGuid);
+        String autoCompleteQuery = request.queryParams(RouteConstants.QueryParam.TYPEAHEAD_QUERY);
+        String queryLimit = request.queryParams(RouteConstants.QueryParam.TYPEAHEAD_QUERY_LIMIT);
+        int limit = (StringUtils.isNotBlank(queryLimit) && Integer.valueOf(queryLimit) <= DEFAULT_LIMIT)
+                ? Integer.valueOf(queryLimit) : DEFAULT_LIMIT;
+        ContentStyle style = RouteUtil.parseContentStyleHeaderOrHalt(request, response, ContentStyle.STANDARD);
+        LanguageDto preferredUserLanguage = RouteUtil.getUserLanguage(request);
+
+        LOG.info("Fetching auto complete picklist options for activity instance {} and participant {} in study {} by operator {} "
+                + "(isStudyAdmin={})", instanceGuid, participantGuid, studyGuid, operatorGuid, isStudyAdmin);
+
+        RemoteAutoCompleteResponse result = TransactionWrapper.withTxn(handle -> {
+
+            //Get all options first from ActivityDef (already in memory) then render and match
+            long timestamp = Instant.now().toEpochMilli();
+            long langCodeId = preferredUserLanguage.getId();
+            List<PicklistOption> suggestions = new ArrayList<>();
+            List<PicklistOption> allOptions = getPicklistOptions(handle, studyGuid, instanceGuid, questionStableId);
+            if (StringUtils.isBlank(autoCompleteQuery)) {
+                LOG.info("Option suggestion query is blank, returning all results size to default limit");
+                suggestions = allOptions.stream().limit(limit).collect(Collectors.toList());
+                renderer.bulkRenderAndApply(handle, allOptions, style, langCodeId, timestamp);
+            } else {
+                renderer.bulkRenderAndApply(handle, allOptions, style, langCodeId, timestamp);
```

Comment:
Render Picklist label templates depending on users current language.
May be caching rendered templates for all possible languages might be an improvement to pursue but will look into it later.